crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -225,13 +228,25 @@ impl Starknet {
         Ok(())
     }
 
+    fn next_block_timestamp(&mut self) -> BlockTimestamp {
+        match self.next_block_timestamp {
+            Some(timestamp) => {
+                self.next_block_timestamp = None;
+                BlockTimestamp(timestamp)
+            }
+            None => BlockTimestamp(
+                (Starknet::get_unix_timestamp_as_seconds() as i64
+                    + self.pending_block_timestamp_shift) as u64,
+            ),
```

Comment:
Pythonic Devnet had three things done in case next block timestamp is `None`:
https://github.com/0xSpaceShard/starknet-devnet/blob/7ea5a3f9a400d59979380c278adda479b6632812/starknet_devnet/block_info_generator.py#L33-L35

Reply:
do you mean in the other case? when it's not `None`?

Reply:
these lines are the same next_block_timestamp is returning block_timestamp that is set later:
```
        // set block timestamp and context block timestamp for contract execution
        let block_timestamp = self.next_block_timestamp();
        new_block.set_timestamp(block_timestamp);
```